controller/appcontroller.go

```diff
@@ -942,6 +942,11 @@ func (ctrl *ApplicationController) autoSync(app *appv1.Application, syncStatus *
 		logCtx.Infof("Skipping auto-sync: deletion in progress")
 		return nil
 	}
+	// Check if there are any active maintenance windows and skip if true
+	active := app.Spec.SyncPolicy.Automated.MaintenanceWindows.Active()
+	if active {
```

Comment:
@alexec 

> What happens I click "sync" in the UI or if I invoke `argocd app sync` and I'm in a maintenance window

If you are in a window and run a manual sync it will still sync, the window only disables auto-sync

> How will I know an app is in its window? Do we want to print that on `argocd app get` and show in the UI when you load the app?

Yes, that is a good idea, I update the cli and ui.